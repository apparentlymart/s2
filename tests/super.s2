#-*-c++-*-

layerinfo "type" = "core";
layerinfo "majorversion" = 0;

class Animal
{
  function out;
}

class Dog extends Animal
{
  var string name;
  var int weight;
}

class KillerDog extends Dog
{
}

function Animal::out {
  println "I'm an animal!";
}

function Dog::out {
  println "I'm a dog named \"$.name\".";
  $super->out();
}

function KillerDog::out {
  println "I'm a killer dog named \"$.name\" with weight $.weight";
  $super->out();
}

function main ()
{
  var KillerDog kd = new KillerDog;
  $kd.name = "Killer Dog";
  $kd.weight = 9999;
  $kd->out();

  var Dog d = $kd;
  "And the same:\n";
  $d->out();
}

