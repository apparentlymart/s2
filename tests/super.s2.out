I'm a killer dog named "Killer Dog" with weight 9999
I'm a dog named "Killer Dog".
I'm an animal!
And the same:
I'm a killer dog named "Killer Dog" with weight 9999
I'm a dog named "Killer Dog".
I'm an animal!

